OpenChange/MAPIStoreContactsMessage.m

```diff
@@ -1373,4 +1375,25 @@ - (void) save:(TALLOC_CTX *) memCtx
   [self updateVersions];
 }
 
+- (void) save:(TALLOC_CTX *) memCtx
+{
+  NSString *messageClass = [properties objectForKey: MAPIPropertyKey(PR_MESSAGE_CLASS_UNICODE)];
+  if (!messageClass) 
+    {
+      [self logWithFormat: @"ERROR: No message class in ContactMessage. Not saving anything"];
+      return;      
+    }
```

Comment:
AFAIK, the *PidTagMessageClass* is not mandatory to be set by the client and, for instance, while updating the MessageClass is likely to be avoided, so I would not include this condition. It is assumed throughout the code the PidTagMessageClass is the one related with that folder, only `MAPIStoreDBFolder` can hold all kinds of messages.